json-path/src/main/java/com/jayway/jsonpath/internal/token/PathToken.java

```diff
@@ -40,7 +45,13 @@ void handleObjectProperty(String currentPath, Object model, EvaluationContextImp
 
         if(properties.size() == 1) {
             String property = properties.get(0);
-            String evalPath = currentPath + "['" + property + "']";
+        	StringBuilder buf = BUFFERS.get() ;
+        	buf.setLength(0);
+        	buf.append(currentPath) ;
+        	buf.append('[') ;
```

Comment:
Tests break by this commit. 

You have missed the ' char wrapping the `currentPath` inside the  brackets.
It should probaly be:
```java
buf.append('[');
buf.append('\'');
buf.append(currentPath);
buf.append('\'');
buf.append(']');
...
```